pkg/config/remote/client.go

```diff
@@ -106,13 +127,17 @@ func (c *Client) startFn() {
 // structure in startFn.
 func (c *Client) pollLoop() {
 	for {
+		interval := c.backoffPolicy.GetBackoffDuration(c.backoffErrorCount)
 		select {
 		case <-c.ctx.Done():
 			return
-		case <-time.After(c.pollInterval):
+		case <-time.After(c.pollInterval + interval):
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 Nitpick:** ‏IIUC we'll be getting a bit more than 2mins here, as interval can get up to 2mins. If you don't want to go over it, we can take the min between this and `maximalMaxBackoffTime`